File: 4.Java/PROYECTO_FINAL_DANIEL_JAY/src/proyecto_final_daniel_jay/Persona.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package proyecto_final_daniel_jay;

/**
 *
 * @author Estudiante
 */
public abstract class Persona {
    private String Nombre;
    private int Cedula;
    private String Direccion;
    private int telefono;
    private int DiaNacimiento;
    private int MesNacimiento;
    private int AñoNacimiento;
    //Contructor
    public Persona(String Nombre, int Cedula, String Direccion,int Telefono, int DiaNac, int MesNac, int AñoNac){
    this.Nombre=Nombre;
    this.Cedula=Cedula;
    this.Direccion=Direccion;
    this.telefono=Telefono;
    this.DiaNacimiento=DiaNac;
    this.MesNacimiento=MesNac;
    this.AñoNacimiento=AñoNac;
    }
    //Metodo Calcular edad
    public String CalularEdad(){
    //La fecha esta calculada tomando como referencia el 10 de noviembre del 2023
    int edad = 2023-this.AñoNacimiento;
    if(this.MesNacimiento>10){
    if(this.DiaNacimiento>10){
    edad = edad-1;
    }
    }
    return "La edad de "+this.Nombre+" es: "+edad;
    }
    
    //Getter and Setter

    public String getNombre() {
        return Nombre;
    }

    public void setNombre(String Nombre) {
        this.Nombre = Nombre;
    }

    public int getCedula() {
        return Cedula;
    }

    public void setCedula(int Cedula) {
        this.Cedula = Cedula;
    }

    public String getDireccion() {
        return Direccion;
    }

    public void setDireccion(String Direccion) {
        this.Direccion = Direccion;
    }

    public int getTelefono() {
        return telefono;
    }

    public void setTelefono(int telefono) {
        this.telefono = telefono;
    }

    public int getDiaNacimiento() {
        return DiaNacimiento;
    }

    public void setDiaNacimiento(int DiaNacimiento) {
        this.DiaNacimiento = DiaNacimiento;
    }

    public int getMesNacimiento() {
        return MesNacimiento;
    }

    public void setMesNacimiento(int MesNacimiento) {
        this.MesNacimiento = MesNacimiento;
    }

    public int getAñoNacimiento() {
        return AñoNacimiento;
    }

    public void setAñoNacimiento(int AñoNacimiento) {
        this.AñoNacimiento = AñoNacimiento;
    }
    
    }

File: 8.C#/27.EJERCICIOS/ejercicio19/Persona.cs
namespace coreObjectOrientedConcepts {
    internal abstract class Persona {
        public string Nombre;
        public int Edad;
        public int Cedula;
        public int Telefono;
        public string Correo;

        //Constructor default 
        public Persona(){
            Nombre = "NA";
            Edad = 0;
            Cedula = 0;
            Telefono = 0;
            Correo = "NA";
        }
        //Constructor con parametros
        public Persona(string Nombre, int Edad, int Cedula, int Telefono, string Correo){
            this.Nombre = Nombre;
            this.Edad = Edad;
            this.Cedula = Cedula;
            this.Telefono = Telefono;
            this.Correo = Correo;
        }

        public virtual void DatosUsuario(){
            Console.WriteLine($"Nombre: {Nombre}");
            Console.WriteLine($"Edad: {Edad}");
            Console.WriteLine($"Cedula: {Cedula}");
            Console.WriteLine($"Telefono: {Telefono}");
            Console.WriteLine($"Correo: {Correo}");
        }
    }
}